Xamarin.Essentials/Launcher/Launcher.shared.cs

```diff
@@ -36,5 +37,81 @@ public static Task OpenAsync(Uri uri)
 
             return PlatformOpenAsync(uri);
         }
+
+        public static Task OpenAsync(OpenFileRequest request)
+        {
+            ExperimentalFeatures.VerifyEnabled(ExperimentalFeatures.ShareFileRequest);
+
+            return PlatformOpenAsync(request);
+        }
+    }
+
+    public class OpenFileRequest
+    {
+        public OpenFileRequest()
+        {
+            ExperimentalFeatures.VerifyEnabled(ExperimentalFeatures.OpenFileRequest);
+        }
+
+        public OpenFileRequest(string title, LauncherFile file)
+        {
+            ExperimentalFeatures.VerifyEnabled(ExperimentalFeatures.OpenFileRequest);
+            Title = title;
+            File = file;
+        }
+
+        public OpenFileRequest(string title, FileBase file)
+        {
+            ExperimentalFeatures.VerifyEnabled(ExperimentalFeatures.OpenFileRequest);
+            Title = title;
+            File = new LauncherFile(file);
+        }
+
+        public string Title { get; set; }
+
+        public LauncherFile File { get; set; }
+    }
+
+    public class LauncherFile : FileBase
+    {
+        public LauncherFile(string fullPath)
+            : base(fullPath)
+        {
+            ExperimentalFeatures.VerifyEnabled(ExperimentalFeatures.OpenFileRequest);
+        }
+
+        public LauncherFile(string fullPath, string contentType)
+            : base(fullPath, contentType)
+        {
+            ExperimentalFeatures.VerifyEnabled(ExperimentalFeatures.OpenFileRequest);
+        }
+
+        public LauncherFile(FileBase file)
+            : base(file)
+        {
+            ExperimentalFeatures.VerifyEnabled(ExperimentalFeatures.OpenFileRequest);
+        }
+
+        string attachmentName;
+
+        public string AttachmentName
```

Comment:
I see this is the third time we have this implemented (Share and Email). We should probably move the implementation to the base (and make it protected if we don't want it visible). We can call it something like `FileNameOverride` and the implementations can just use this value under the hood.

But, I don't see this being used here at all, so I am not sure if this even needs to be here for this case.

Reply:
Fixed!